contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,370 @@
+        )
+        # A file be treated as the reference file while doing PSNR
+        # Comparision. If you want to use other file as reference while doing
+        # PSNR comparison, please reassign the full path to it
+        self._psnr_reference_file = get_golden_sample(
+            width=self._width_to,
+            height=self._height_to,
+            framerate=self._framerate,
+        )
+        self._output_file_full_path = os.path.join(
+            PLAINBOX_SESSION_SHARE,
+            self._output_file_name,
+        )
+
+    @abstractmethod
+    def _build_command(self) -> str:
+        """
+        Execute the command associated with the given method name for a
+        platform.
+        Returns:
+            str: The corresponding command or an error message if the method
+            is not found.
+        """
+        pass
+
+    def execute_encode_command(self) -> None:
+        logging.debug("Executing Encode Command...")
+        execute_command(self._build_command())
+
+    def compare_psnr(self) -> None:
+        logging.info(
+            "Compare the PSNR: {} vs {}".format(
+                self._psnr_reference_file, self._output_file_full_path
+            )
+        )
+        avg_psnr, _ = get_average_psnr(
+            self._psnr_reference_file, self._output_file_full_path
+        )
+        logging.info("Average PSNR: {}".format(avg_psnr))
+        if avg_psnr < 30 and avg_psnr > 0:
+            raise SystemExit(
+                "Error: The average PSNR value did not reach the acceptable"
+                " threshold (30 dB)"
+            )
+        logging.info("Pass: Average PSNR meets the acceptable threshold")
+
+    def check_metadata(self) -> None:
+        logging.debug("Checking metadata...")
+        outcome = execute_command(
+            cmd="gst-discoverer-1.0 {}".format(self._output_file_full_path)
+        )
+
+        # Check the meta
+        is_metadata_good = True
+        p = self._extract_metadata_property(input=outcome)
+        if not p.get("width") or p.get("width") != self._width_to:
+            logging.error(
+                "expect width is '{}' but got '{}'".format(
+                    self._width, p.get("width")
+                )
+            )
+            is_metadata_good = False
+        if not p.get("height") or p.get("height") != self._height_to:
+            logging.error(
+                "expect height is '{}' but got '{}'".format(
+                    self._height, p.get("height")
+                )
+            )
+            is_metadata_good = False
+
+        if not is_metadata_good:
+            raise SystemError("Error: Checking metadata failed")
+
+    def _extract_metadata_property(self, input: str) -> Dict:
+        properties = {}
+        width_pattern = re.compile(r"Width: (\d+)")
+        height_pattern = re.compile(r"Height: (\d+)")
+
+        # Check if either video or video(image) pattern matches
+        properties["width"] = (
```

Comment:
If you change the argument parser type for width_to and width_from to `int`, make sure to call `int()` here